src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecLoaderExtensions.cs

```diff
@@ -16,13 +16,45 @@
 
 using System;
 using System.IO;
+using System.Linq;
+using System.Reflection;
 using System.Text;
+using Nethermind.Config;
 using Nethermind.Logging;
 
 namespace Nethermind.Specs.ChainSpecStyle
 {
     public static class ChainSpecLoaderExtensions
     {
+
+        public static ChainSpec LoadFromEmbeddedOrFile(this IChainSpecLoader chainSpecLoader, string fileName, ILogger logger)
+        {
+            string resourceName = fileName;
+            if (!resourceName.Contains('/'))
+            {
+                resourceName = "chainspec/" + resourceName;
+            }
+            if (!resourceName.EndsWith(".json", StringComparison.InvariantCultureIgnoreCase))
+            {
+                resourceName += ".json";
+            }
+            resourceName = resourceName.Replace('/', '.');
+            Assembly assembly = typeof(IConfig).Assembly;
+            string[] embeddedChainSpecFiles = assembly.GetManifestResourceNames();
+            if (!embeddedChainSpecFiles.Any(s => s.EndsWith(resourceName)))
```

Comment:
Please provide `StringComparison.Ordinal` or `StringComparison.OrdinalIgnoeCase` to `EndsWith()` as above.